src/class-parsely.php

```diff
@@ -653,4 +670,93 @@ private function are_credentials_managed(): bool {
 
 		return true;
 	}
+
+	/**
+	 * Sets the values of managed options.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 */
+	private function set_managed_options(): void {
+		$managed_options = apply_filters( 'wp_parsely_managed_options', false );
+
+		if ( ! is_array( $managed_options ) ) {
+			return;
+		}
+
+		// Don't allow certain options to be set as managed.
+		unset(
+			$managed_options['apikey'],
+			$managed_options['api_secret'],
+			$managed_options['metadata_secret'],
+			$managed_options['track_post_types'],
+			$managed_options['track_page_types'],
+			$managed_options['plugin_version']
+		);
+
+		/**
+		 * Current options.
+		 *
+		 * @var Parsely_Options $current_options
+		 */
+		$current_options = get_option( self::OPTIONS_KEY, array() );
+
+		// Set managed options values.
+		foreach ( $managed_options as $key => $value ) {
+			$is_option_valid = isset( $this->option_defaults[ $key ] );
+
+			if ( $is_option_valid ) {
+				if ( null === $value ) {
+					// When null, the option gets its value from the database.
+					$this->managed_options[ $key ] =
+						$current_options[ $key ] ?? $this->option_defaults[ $key ];
+				} else {
+					$this->managed_options[ $key ] =
+						$this->sanitize_managed_option( $key, $value );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Sanitizes the value of the passed managed option.
+	 *
+	 * @since 3.9.0
+	 * @access private
+	 *
+	 * @param string      $option_id The option's ID.
+	 * @param bool|string $value The option's value.
+	 *
+	 * @return bool|string The sanitized option value.
+	 */
+	private function sanitize_managed_option( string $option_id, $value ) {
+		$option_value_type = gettype( $this->option_defaults[ $option_id ] );
+
+		if ( 'boolean' === $option_value_type && ! is_bool( $value ) ) {
+			return false;
```

Comment:
We should trigger a `_doing_it_wrong` here.

Reply:
Done in https://github.com/Parsely/wp-parsely/pull/1724/commits/6d7681ce2b94ede00534d7505220f97f8944290c with some fixes in https://github.com/Parsely/wp-parsely/pull/1724/commits/9789500925076c9f3138661db208f01a89a44021. Will mark similar requests as resolved.